src/rpc/handlers/LedgerEntry.cpp

```diff
@@ -157,21 +157,29 @@ LedgerEntryHandler::process(LedgerEntryHandler::Input input, Context const& ctx)
         return Error{*status};
 
     auto const lgrInfo = std::get<ripple::LedgerHeader>(lgrInfoOrStatus);
-    auto const ledgerObject = sharedPtrBackend_->fetchLedgerObject(key, lgrInfo.seq, ctx.yield);
-
-    if (!ledgerObject || ledgerObject->empty())
-        return Error{Status{"entryNotFound"}};
-
+    std::optional<data::Blob> ledgerObject;
+    auto output = LedgerEntryHandler::Output{};
+    output.deletedLedgerIndex = std::nullopt;
+    ledgerObject = sharedPtrBackend_->fetchLedgerObject(key, lgrInfo.seq, ctx.yield);
+    if (!ledgerObject || ledgerObject->empty()) {
+        if (input.includeDeleted == false) {
+            return Error{Status{"entryNotFound"}};
+        }
+        ledgerObject = sharedPtrBackend_->fetchLedgerObject(key, lgrInfo.seq - 1, ctx.yield);
+        if (!ledgerObject || ledgerObject->empty()) {
+            return Error{Status{"entryNotFound"}};
+        }
+        output.deletedLedgerIndex = lgrInfo.seq;
+    }
+    
     ripple::STLedgerEntry const sle{ripple::SerialIter{ledgerObject->data(), ledgerObject->size()}, key};
 
     if (input.expectedType != ripple::ltANY && sle.getType() != input.expectedType)
         return Error{Status{"unexpectedLedgerType"}};
 
-    auto output = LedgerEntryHandler::Output{};
     output.index = ripple::strHex(key);
     output.ledgerIndex = lgrInfo.seq;
     output.ledgerHash = ripple::strHex(lgrInfo.hash);
-
```

Comment:
Please don't remove spaces that make sense - they separate blocks of variables from `if`s, unrelated `if`s from each other, etc. - let's add them back and also add some more around your 'if' statements 👍